src/item.cpp

```diff
@@ -197,25 +197,15 @@ item::item(JsonObject &jo)
     deserialize(jo);
 }
 
-item::~item()
-{
-}
-
-void item::make( const std::string new_type, bool scrub )
+item& item::convert( const itype_id& new_type )
 {
     type = find_type( new_type );
-    contents.clear();
-
-    if (scrub) {
-        components.clear();
-        charges = -1;
-        bday = 0;
-        name = "";
-        curammo = NULL;
+    if( is_null() ) {
```

Comment:
`is_null` is for the null item check, nothing else. The null item is an "uninitialized" item, for example one create via the default constructor. Invalid item ids should never be encountered, if they are encountered, the `item_factory` will recognize them and issue the debug message. 

This provides a nice and central place for that check, all the other code can assume that `find_template` returns a valid pointer and doesn't need to care about invalid item ids.